src/main/java/gregtech/api/metatileentity/IMiner.java

```diff
@@ -1,53 +1,72 @@
+        BASIC(16, 3, 3, GTUtility.romanNumeralString(3), 8),
+        LARGE(4, 5, 3, GTUtility.romanNumeralString(3), 16),
+        ADVANCE(1, 7, 3, GTUtility.romanNumeralString(3), 32);
+
+        public final int tick;
+        public final int radius;
+        public final int fortune;
+        public final int drillingFluidConsumePerTick;
+        public final String fortuneString;
+
+
+        Type(int tick, int radius, int fortune, String fortuneString, int drillingFluidConsumePerTick) {
+            this.tick = tick;
+            this.radius = radius;
+            this.fortune = fortune;
+            this.drillingFluidConsumePerTick = drillingFluidConsumePerTick;
+            this.fortuneString = fortuneString;
+        }
+
     }
 
-    static List<BlockPos> getBlockToMinePerChunk(IMiner miner, AtomicLong x, AtomicLong y, AtomicLong z, ChunkPos chunkPos) {
-        List<BlockPos> blocks = new ArrayList<>();
-        if ((miner instanceof MetaTileEntity)) {
-            MetaTileEntity mteMiner = (MetaTileEntity) miner;
-            for (int i = 0; i < miner.getNbBlock(); i++) {
-                if (y.get() >= 0 && mteMiner.getOffsetTimer() % miner.getTicksPerOperation() == 0) {
-                    if (z.get() <= chunkPos.getZEnd()) {
-                        if (x.get() <= chunkPos.getXEnd()) {
-                            BlockPos blockPos = new BlockPos(x.get(), y.get(), z.get());
-                            Block block = mteMiner.getWorld().getBlockState(blockPos).getBlock();
-                            if (mteMiner.getWorld().getTileEntity(blockPos) == null) {
-                                if (GTUtility.isOre(block)) {
-                                    blocks.add(blockPos);
-                                }
-                            }
-                            x.incrementAndGet();
-                        } else {
-                            x.set(chunkPos.getXStart());
-                            z.incrementAndGet();
+    Type getType();
+
+    World getWorld();
+
+    long getOffsetTimer();
+
+    static LinkedList<BlockPos> getBlocksToMine(IMiner miner, AtomicInteger x, AtomicInteger y, AtomicInteger z, AtomicInteger startX, AtomicInteger startZ, AtomicInteger startY, int aRadius) {
+        LinkedList<BlockPos> blocks = new LinkedList<>();
+        while (y.get() > 0) {
+            if (z.get() <= startZ.get() + aRadius * 2) {
+                if (x.get() <= startX.get() + aRadius * 2) {
+                    LinkedList<BlockPos> blockPos = new LinkedList<>();
+                    for (int a = 0; a < aRadius * 2; a++) {
+                        blockPos.addLast(new BlockPos(x.get(), y.get(), startZ.get() + a));
+                        Block block = miner.getWorld().getBlockState(blockPos.getLast()).getBlock();
+                        if (miner.getWorld().getTileEntity(blockPos.getLast()) == null && GTUtility.isOre(block)) {
+                            blocks.addLast(blockPos.getLast());
```

Comment:
Scanning is  significant complexity, but I don't see why it is necessary to save scanning results in `blockPos`. Why not just check inside the loop to see if the block is ore and add it to `blocks`?

Reply:
True, fixed and committed.